hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/table/functional/TestHoodieSparkCopyOnWriteTableRollbackTableVersionSix.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.table.functional;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import static org.apache.hudi.common.model.HoodieTableType.COPY_ON_WRITE;
+
+@Tag("functional")
+public class TestHoodieSparkCopyOnWriteTableRollbackTableVersionSix extends TestHoodieSparkRollback {
```

Comment:
Could you report the time to run all the new tests added in this PR both locally and in CI (search in the `Tests` tab shown below)?
<img width="1442" alt="Screenshot 2025-03-27 at 16 23 04" src="https://github.com/user-attachments/assets/d6446b75-bd80-488b-9ac2-fde97a1fbfd6" />

Reply:
Locally
TestCommitMetadataResolver - 2s
TestHoodieSparkCopyOnWriteTableRollbackTableVersionSix - 36s
TestHoodieSparkMergeOnReadTableCompaction#testCompactionSchedulingWithUncommittedLogFileOrRollback - 30s
new tests in TestHoodieSparkMergeOnReadTableRollback - 51s
org.apache.hudi.table.marker.TestWriteMarkersBase#testGetAppendedLogPaths - 18s

Reply:
from CI runs : 
```
TestCommitMetadataResolver:
  testReconcileMetadataForMissingFiles - 400ms
  testGetPathInfoUnderPartition - 1s
TestHoodieSparkCopyOnWriteTableRollbackTableVersionSix: - could not get test run time from CI
TestHoodieSparkMergeOnReadTableCompaction#testCompactionSchedulingWithUncommittedLogFileOrRollback - 
TestHoodieSparkMergeOnReadTableRollback - Could not get test run time from CI. 
org.apache.hudi.table.marker.TestWriteMarkersBase#testGetAppendedLogPaths - 934ms.
```

Reply:
Also, ran two new tests added recently locally.

TestDirectWriteMarkersV1 : 10s
TestTimelineServerBasedWriteMarkersV1: 18s